consensus/src/dag/dag_handler.rs

```diff
@@ -90,10 +107,19 @@ impl NetworkHandler {
                 .verify(&self.epoch_state.verifier)
                 .and_then(|_| self.node_receiver.process(node))
                 .map(|r| r.into()),
-            DAGMessage::CertifiedNodeMsg(certified_node_msg) => certified_node_msg
-                .verify(&self.epoch_state.verifier)
-                .and_then(|_| self.dag_driver.process(certified_node_msg.certified_node()))
-                .map(|r| r.into()),
+            DAGMessage::CertifiedNodeMsg(certified_node_msg) => {
+                match certified_node_msg.verify(&self.epoch_state.verifier) {
+                    Ok(_) => match self.state_sync_trigger.check(certified_node_msg).await {
+                        ret @ (NeedsSync(_), None) => return Ok(ret.0),
+                        (Synced, Some(certified_node_msg)) => self
```

Comment:
the message can be carried in the StateSyncStatus::Synced to avoid the second Option?

Reply:
I need to send the Synced status from process_rpc fn as well, so I either clone the ceritifed_node_msg or use the second option. 

Alternatively, I could have two enums, state sync check return one enum and I can convert it to another one for process_rpc. i thought it was too much.

Reply:
I don't think we need to define two enum, this function can just return a Result<(), CertifiedNodeMessage>?